components/iam/pkg/oidc/service.go

```diff
@@ -87,6 +95,25 @@ func (s *Service) GetStartParams(config *ClientConfig) (*StartParams, error) {
 	}, nil
 }
 
+// TODO(at) state should be a JWT encoding a redirect location
+// For now, just use base64
+func encodeStateParam(state StateParam) (string, error) {
+	var buf bytes.Buffer
+	encoder := base64.NewEncoder(base64.StdEncoding, &buf)
```

Comment:
here the task was to serialize the state into base64 encoded string. I didn't find a way to create a writer form a the default encoder, which seems to be reasonable.

please bear with me if this is not ideal, but that simple serialization isn't here to stay. it needs to be replaced by JWT before this will be GA.

Reply:
I'm saying you can replace this with
```
b, err := json.Marshal(state)
	if err != nil {
		return "", fmt.Errorf("failed to marshal state to json: %w", err)
	}
	
	return base64.StdEncoding.EncodeToString(b), nil
```

which is simpler

Reply:
ok